src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/integration/UsersTest.kt

```diff
@@ -515,6 +515,57 @@ class UsersTest : InitialiseDatabasePerClassTestBase() {
       }
     }
 
+    @ParameterizedTest
+    @EnumSource(value = UserRole::class, names = ["CAS1_ADMIN", "CAS1_WORKFLOW_MANAGER", "CAS1_JANITOR", "CAS1_USER_MANAGER"])
+    fun `GET to users with an approved role returns list filtered by user's CRU management area`(role: UserRole) {
+      `Given a User`(roles = listOf(UserRole.CAS1_MATCHER)) { matcher, _ ->
+        `Given a User`(roles = listOf(UserRole.CAS1_MANAGER)) { manager, _ ->
+          `Given a User` { userWithNoRole, _ ->
+            `Given a User`(roles = listOf(role)) { requestUser, jwt ->
+              val apArea = `Given an AP Area`()
+
+              val probationRegionApArea = `Given an AP Area`()
+              val probationRegion = probationRegionEntityFactory.produceAndPersist {
+                withApArea(probationRegionApArea)
+              }
+
+              val userOne = userEntityFactory.produceAndPersist {
+                withProbationRegion(probationRegion)
+                withApArea(apArea)
+                withCruManagementArea(apArea.defaultCruManagementArea)
+              }
+
+              val userTwo = userEntityFactory.produceAndPersist {
+                withProbationRegion(probationRegion)
+                withApArea(apArea)
+                withCruManagementArea(apArea.defaultCruManagementArea)
+              }
+
+              webTestClient.get()
+                .uri("/users?cruManagementAreaId=${apArea.defaultCruManagementArea.id}")
```

Comment:
Does it make sense to use the `defaultCruManagementArea` here?

Reply:
I think it may be cleaner if you create some CRU Management Areas yourself and assign these directly to the user's cruManagementArea property, ignoring ap areas completely e.g. 

```
val cruManagementArea1 = `Given a CAS1 CRU Management Area`()
val cruManagementArea2 = `Given a CAS1 CRU Management Area`()

...

userEntityFactory.produceAndPersist {
   ...
   withCruManagementArea(cruManagementArea1)
}
```

(rough code ... i think it will compile!)

Reply:
No problem, done this.